modules/contact/src/constraints/ComputeFrictionalForceLMMechanicalContact.C

```diff
@@ -0,0 +1,158 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComputeFrictionalForceLMMechanicalContact.h"
+#include "DisplacedProblem.h"
+#include "Assembly.h"
+
+#include "DualRealOps.h"
+
+registerMooseObject("ContactApp", ComputeFrictionalForceLMMechanicalContact);
+
+InputParameters
+ComputeFrictionalForceLMMechanicalContact::validParams()
+{
+  InputParameters params = ComputeWeightedGapLMMechanicalContact::validParams();
+  params.addClassDescription("Computes the tangential frictional forces");
+  params.addRequiredCoupledVar("friction_lm", "The frictional Lagrange's multiplier");
+  params.addParam<Real>("c_t", 1e0, "Numerical parameter for tangential constraints");
```

Comment:
I actually can't remember if we ended up bumping the default value for `c` for normal contact but I think I remember you thinking that we should. Do you think *this* default is appropriate?

Reply:
The few problems I have run converge well with low `c_t`. So, it is intended to be lower than `c` but maybe subject to revision in the future. 

You did increase the value of `c_n` in the base class to 1.0e6, which I think it's reasonable. Some authors recommend it to be on the order the Young's modulus so we may need to increase it when we deal with steel, we'll see.

Reply:
We can make  some sensitivity analysis on those parameters time permitting.